### ISSUE TITLE
Inline annotations to ignore a rule for a specific instance

### ISSUE THREAD

**Issue by swick** [bilelmoussaoui/gobject-lint#45]
The rule use_g_strlcpy caught this:
```c
  XdpInvalidateData *data = g_malloc0 (sizeof (XdpInvalidateData) + strlen (name) + 1);
  data->parent_ino = parent->ino;
  strcpy (data->name, name);
```
In general it is a really good rule that I would like to not turn off, but in this case, the code is correct and making the linter happy would only require complicating things.

Maybe it would make sense to allow annotations inline to ignore a rule in a specific case, something like

```c
  XdpInvalidateData *data = g_malloc0 (sizeof (XdpInvalidateData) + strlen (name) + 1);
  data->parent_ino = parent->ino;
  /* goblint-ignore: use_g_strlcpy */
  strcpy (data->name, name);
```

**Comment by swick** [comment]
Similar issue: I have a header file and corresponding codegen, so `missing_implementation` gets triggered.

```
/workspace/src/xdp-method-info.h:36:1
    int option_arg;
  } XdpMethodInfo;
  ...
  const XdpMethodInfo *
  xdp_method_info_find (const char *interface,
                        const char *method);
  ...
  >const XdpMethodInfo *
  xdp_method_info_get_all (void);
  ...
  unsigned int
  error: Function 'xdp_method_info_get_all' is declared in a header but has no implementation
  rule: missing_implementation
```

**Comment by bilelmoussaoui** [comment]
You can do that now with https://github.com/bilelmoussaoui/goblint/blob/main/CONFIG.md#inline-ignore-directives. Please let me know if that is enough for your use cases

**Comment by swick** [comment]
Yup, managed to get rid of the remaining violations, e.g. https://github.com/flatpak/xdg-desktop-portal/pull/1974/changes/bbd8d2290b67076d1b46c7723ee10c7f96f961e4#diff-2f05bdfc1f7bed1308fa7848357777aad2cadd1d6133d92678296ef2697c4e49R1829
